ignite/engine/deterministic.py

```diff
@@ -192,7 +193,14 @@ def state_dict(self) -> OrderedDict:
     def _init_run(self) -> None:
         self.state.seed = int(torch.randint(0, int(1e9), (1,)).item())
         if torch.cuda.is_available():
-            torch.backends.cudnn.deterministic = True
+            if hasattr(torch, "use_deterministic_algorithms"):
+                # according to https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
+                # CUBLAS_WORKSPACE_CONFIG must be set to let cuBLAS behave deterministic.
+                # **the behavior is expected to change in a future release of cuBLAS**.
+                os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
+                torch.use_deterministic_algorithms(True)
```

Comment:
Maybe, we should set `warn_only=True` such that we do not break previous code but only raise warnings about non-deterministic implementation.